cmd/juju/application/deploy.go

```diff
@@ -469,105 +474,126 @@ use
 
 Only top level machines can be mapped in this way, just as only top level
 machines can be defined in the machines section of the bundle.
 
+Further reading: https://docs.jujucharms.com/stable/charms-deploying
 
 Examples:
-    juju deploy mysql               (deploy to a new machine)
-    juju deploy mysql --to 23       (deploy to preexisting machine 23)
-    juju deploy mysql --to lxd      (deploy to a new LXD container on a new machine)
-    juju deploy mysql --to lxd:25   (deploy to a new LXD container on machine 25)
-    juju deploy mysql --to 24/lxd/3 (deploy to LXD container 3 on machine 24)
+
+Deploy to a new machine:
+
+    juju deploy apache2
+
+Deploy to machine 23:
+
+    juju deploy mysql --to 23
+
+Deploy to a new LXD container on a new machine:
+
+    juju deploy mysql --to lxd
+
+Deploy to a new LXD container on machine 25:
+
+    juju deploy mysql --to lxd:25
+
+Deploy to LXD container 3 on machine 24:
+
+    juju deploy mysql --to 24/lxd/3
+
+Deploy 2 units, one on machine 3 and one to a new LXD container on machine 5:
 
     juju deploy mysql -n 2 --to 3,lxd:5
-    (deploy 2 units, one on machine 3 & one to a new LXD container on machine 5)
+
+Deploy 3 units, one on machine 3 and the remaining two on new machines:
 
     juju deploy mysql -n 3 --to 3
-    (deploy 3 units, one on machine 3 & the remaining two on new machines)
 
-    juju deploy mysql -n 5 --constraints mem=8G
-    (deploy 5 units to machines with at least 8 GB of memory)
+Deploy to a machine with at least 8 GiB of memory:
+
+    juju deploy postgresql --constraints mem=8G
+
+Deploy to a specific availability zone (provider-dependent):
 
     juju deploy mysql --to zone=us-east-1a
-    (provider-dependent; deploy to a specific AZ)
+
+Deploy to a specific MAAS node:
 
     juju deploy mysql --to host.maas
-    (deploy to a specific MAAS node)
+
+Deploy to a machine that is in the 'dmz' network space but not in either the
+'cms' nor the 'database' spaces:
 
     juju deploy haproxy -n 2 --constraints spaces=dmz,^cms,^database
-    (deploy 2 units to machines that are in the 'dmz' space but not of
-	the 'cmd' or the 'database' spaces)
 
-	juju deploy mycharm --device bitcoinminer=1,nvidia.com/gpu
-	(deploy mycharm requires any Nvidia GPU without needing to further specify any tags)
+Deploy a Kubernetes charm that requires a single Nvidia GPU:
+
+    juju deploy mycharm --device miner=1,nvidia.com/gpu
 
-	juju deploy mycharm --device bitcoinminer=nvidia.com/gpu
-	(deploy mycharm requires any Nvidia GPU. No count is specified, it is assumed to be 1)
+Deploy a Kubernetes charm that requires two Nvidia GPUs that have an
+attribute of 'gpu=nvidia-tesla-p100':
 
-	juju deploy mycharm --device bitcoinminer=1,nvidia.com/gpu,gpu=nvidia-tesla-p100;attr2=attr2
-	(deploy mycharm requires 1*nvidia.com/gpu with attributes: gpu=nvidia-tesla-p10 && attr2=attr2)
+    juju deploy mycharm --device \
+       twingpu=2,nvidia.com/gpu,gpu=nvidia-tesla-p100
```

Comment:
Here's that pesky `--device` thing again.  I must be missing something, but this is what I get on 2.4.4:
```
$ juju deploy mycharm --device miner=1,nvidia.com/gpu
ERROR flag provided but not defined: --device
```

Reply:
2.5 only.